File: _includes/code/basic_java/Greet.java
public class Greet {
    private final String name;
    public Greet(String name) {
	    this.name = name;
    }

    public void greet() {
	    System.out.println("Hello " + this.name);
    }

    public static void main(String[] args) {
	    Greet g = new Greet(args[0]);
	    g.greet();
    }
}

File: zzzz/src/greet.clj
(ns greet)

(gen-class
 :name "greet.Greeter"
 :prefix "greeter-"
 :state "message"
 :init "init"
 :constructors {[String] []}
 :methods [[greet [] void]])

(defn- greeter-init [message]
  [[] message])

(defn- greeter-greet [this]
  (println (str "Hello " (.message this) "!")))

(gen-class
 :name "greet.main"
 :prefix "-"
 :main true)

(defn -main [& args]
  (let [g (greet.Greeter. (or (first args) "world"))]
    (.greet g)))
